src/main/java/org/opentripplanner/updater/trip/TimetableSnapshotSource.java

```diff
@@ -291,7 +291,8 @@ public UpdateResult applyTripUpdates(
               tripId,
               serviceDate
             );
-            case CANCELED -> handleCanceledTrip(tripId, serviceDate);
+            case CANCELED -> handleCanceledTrip(tripId, serviceDate, false);
+            case DELETED -> handleCanceledTrip(tripId, serviceDate, true);
```

Comment:
We talked about this in the dev meeting today. I'm not a huge fan of passing booleans to methods. I think I have a slight preference for using an enum instead in this case but will leave the choice up to you: the enum also not very pretty.